sklearn/neighbors/base.py

```diff
@@ -514,13 +578,33 @@ class from an array representing our data set and ask who's
             results = self._tree.query_radius(X, radius,
                                               return_distance=return_distance)
             if return_distance:
-                ind, dist = results
-                return dist, ind
-            else:
-                return results
+                results = results[::-1]
         else:
             raise ValueError("internal: _fit_method not recognized")
 
+        if not query_is_train:
+            return results
+        else:
+            if return_distance:
+                dist, neigh_ind = results
+            else:
+                neigh_ind = results
+
+            for ind, ind_neighbor in enumerate(neigh_ind):
+                mask = ind_neighbor != ind
+                # Corner case if the number of duplicates is more than the number
```

Comment:
This case is inapplicable. There's no such thing as "number of neighbors", and the query index should always be there (at least for eps > 0, and = 0 given an appropriate resolution of #4072 ). (I'd be very surprised if the line in the `if` below has test coverage!!!)